dali/test/dali_operator_test.h

```diff
@@ -23,30 +23,197 @@
+ */
 class DaliOperatorTest : public ::testing::Test, public ::testing::WithParamInterface<Arguments> {
  public:
-  using Verify = std::function<void(TensorListWrapper, TensorListWrapper, Arguments)>;
+  DaliOperatorTest(size_t batch_size, size_t num_thread) :
+          batch_size_(batch_size), num_threads_(num_thread) {}
+
+
+  /**
+   * Type of the function used for verification, whether graph test passed or failed.
+   * Within this function, please use GTest's ASSERTs and EXPECTs, for test validation.
+   *
+   * First argument of function contains all inputs, that have been provided;
+   * Second argument are all outputs from pipeline;
+   * Third argument contains Arguments, with which the pipeline was called
+   *
+   * This is some sort or template method pattern.
+   */
+  using Verify = std::function<void(const std::vector<TensorListWrapper> /* inputs */,
+                                    const std::vector<TensorListWrapper> /* outputs */,
+                                    const Arguments)>;
+
+  /**
+   * Convenient variation of Verify, which is applied for single-input/single-output (Siso) graph
+   */
+  using VerifySiso = std::function<void(const TensorListWrapper /* single input */,
+                                        const TensorListWrapper /* single output */,
+                                        const Arguments)>;
 
  protected:
-  void RunTest(const TensorListWrapper &input, const TensorListWrapper &output,
-               const Arguments &operator_arguments, const Verify &verify) {
+  /**
+   * Runs defined test.
+   *
+   * Call this method within GTest's testing macro (TEST_F, TEST_P, etc...).
+   * The method will set up a testing pipeline,
+   * push data through it and call output verification routines.
+   * Every RunTest call will set up its own pipeline.
+   *
+   * @tparam OutputBackend
+   * @param inputs all inputs to the pipeline
+   * @param outputs placeholder for outputs from pipeline
+   * @param operator_arguments arguments, with which the pipeline will be called
+   * @param verify function, that will be used for test verification
+   */
+  template<typename OutputBackend>
+  void
+  RunTest(const std::vector<TensorListWrapper> &inputs, std::vector<TensorListWrapper> &outputs,
+          const Arguments &operator_arguments, const Verify &verify) {
+    // TODO(mszolucha) implement
   }
 
 
-  void RunTest(const std::vector<TensorListWrapper> &inputs,
-               const std::vector<TensorListWrapper> &outputs,
-               const Arguments &operator_arguments, const std::vector<Verify> &verify) {
+  /**
+   * Convenient overload, for specific, single-input/single-output (Siso) graph
+   */
+  template<typename OutputBackend>
+  void RunTest(const TensorListWrapper &input, TensorListWrapper &output,
+               const Arguments &operator_arguments, const VerifySiso &verify) {
+    std::string output_backend = detail::BackendStringName<OutputBackend>();
```

Comment:
This function should basically call:
`RunTest<OutputBackend>({ input }, { output }, arguments, verify_single_io(verify));`

Reply:
We decided, that for now we'll keep SingleIo implementation. When MultiIo implementation is ready, we'll get rid of the old one